### ISSUE TITLE
[BUG] Attach to Unity doesn't stop at breakpoints 

### ISSUE THREAD

**Issue by WardenGnaw** [microsoft/vscode-dotnettools#783]
### Describe the Issue

Originally from https://github.com/dotnet/vscode-csharp/issues/6710 by @COOLIRON2311

## Additional context
It does work without an issue if I attach the debugger via `Attach Unity Debugger` option in Command Palette.
I've tried different versions of all listed extensions with no success. `Attach to Unity` never managed to properly handle breakpoints. However I recall that it was able to do so about two months ago.

### Steps To Reproduce

1. Open any Unity project in Visual Studio Code
2. It atomatically creates `launch.json` with the following contents:
``` json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Attach to Unity",
            "type": "vstuc",
            "request": "attach",
        }
     ]
}
```
3. Switch to debug mode in Unity
4. Start `Attach to Unity` debug task
5. A list of threads is shown in vs code's debugger and "debugger attached" (blue bug) status is indicated by Unity editor.

### Expected Behavior

The debugger stops at breakpoints.

## Actual behavior
The debugger does not stop at breakpoints.

## Environment data
`dotnet --info` output: 
<details>
<summary>Output</summary>

```
ПАКЕТ SDK ДЛЯ .NET:
 Version:   7.0.403
 Commit:    142776d834

Среда выполнения:
 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.403\

Host:
  Version:      7.0.13
  Architecture: x64
  Commit:       3f73a2f186

.NET SDKs installed:
  7.0.403 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.24 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.25 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.25 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```
</details>

VS Code version: `1.84.2`
OS Version: `Windows 11 Home 23H2`
C# Extension: `2.12.19`
C# DevKit: `1.0.14`
Unity Extension: `0.9.3`
Unity: `2022.3.12f1`
Visual Studio Editor package: `2.0.22`

**Comment by jbevain** [comment]
Hi @COOLIRON2311,

Thanks for reporting this. Unfortunately we're unable to reproduce this.

Could you modify your launch.json to enable debugger logs, for instance:

```json
    "configurations": [
        {
            "name": "Attach to Unity",
            "type": "vstuc",
            "request": "attach",
            "logFile": "${workspaceRoot}/debug.log"
        }
```

And send the resulting file to vstusp@microsoft.com?

Thank you!

**Comment by COOLIRON2311** [comment]
@jbevain I've sent the logs just now

**Comment by sailro** [comment]
@COOLIRON2311 given it works when using the Attach dialog but not when using F5 directly, could you sahre the following with us:

- A screenshot of the Attach dialog with all detected processes.
- Your `EditorInstance.json` file that you can find in the `Library` folder of your Unity project. (Unity Editor must be loaded with your project to see this file).

Indeed when you use the Attach dialog, we use the multicast protocol to detect Editors & Players. But when using F5 we use the `EditorInstance.json` file provided by Unity. Perhaps we have an issue here.

Thank you!

**Comment by COOLIRON2311** [comment]
@sailro Sure thing, here you go:
![image](https://github.com/microsoft/vscode-dotnettools/assets/31288825/cc0fb2da-b781-4435-b6e5-d517f30f7d66)

[EditorInstance.json](https://github.com/microsoft/vscode-dotnettools/files/13732596/EditorInstance.json)

**Comment by sailro** [comment]
@COOLIRON2311, in fact we display the TCP port in the dialog, and Unity is using the process id in the `EditorInstance.json`, so it seems to match given the formula to compute the TCP port is `56000+process_id%1000`, so `56000+11776%1000=56776 `

**Comment by COOLIRON2311** [comment]
@sailro Oh, ok. Yeah, I've checked the permissions. Here they are:
``` powershell
icacls.exe .\EditorInstance.json
.\EditorInstance.json NT AUTHORITY\SYSTEM:(I)(F)
                      BUILTIN\Administrators:(I)(F)
                      TUF-DASH-F15\ivani:(I)(F)
```

Anyway, I think I know what caused this problem. I've rebinded my keys some time ago. `F5` for me is `Debug: Start Without Debugging` and `Debug: Start Debugging` is `F8`. Starting the debugging session using `F8` solves the issue (the debugger stops at breakpoints). I'm sorry that you had to go through all of this because of my inadvertence :D

Thank you for your time.
